src/modules/Builder/hooks/useActivitiesRedirection.ts

```diff
@@ -12,15 +12,14 @@ export const useActivitiesRedirection = () => {
   const navigate = useNavigate();
   const { activity } = useCurrentActivity();
   const isNewApplet = useCheckIfNewApplet();
-
   const { watch } = useFormContext();
 
-  const activities = watch('activities');
-  const shouldRedirect = (isNewApplet || activities?.length > 0) && activityId && !activity;
-
   useEffect(() => {
+    const activities = watch('activities');
```

Comment:
please don't forget to unsubscribe then
https://www.react-hook-form.com/api/useform/watch/